scanner/Makefile

```diff
@@ -0,0 +1,170 @@
+# Store tooling in a location that does not affect the system.
+GOBIN := $(CURDIR)/.gobin
+export GOBIN
+PATH := $(GOBIN):"$(PATH)"
+export PATH
+
+# Set to empty string to echo some command lines which are hidden by default.
+SILENT ?= @
+
+SHELL = /bin/bash
+BASE_DIR = $(CURDIR)
+
+TAG := # make sure tag is never injectable as an env var
+
+ifdef CI
+ifneq ($(NIGHTLY_TAG),)
+	TAG := $(NIGHTLY_TAG)
+else ifneq ($(RELEASE_TAG),)
+	TAG := $(RELEASE_TAG)
+endif
+endif
+
+ifeq ($(TAG),)
+	TAG = $(shell git describe --tags --abbrev=10 --dirty --long --exclude '*-nightly-*')
+endif
+
+GO_VERSION := $(shell cat ../EXPECTED_GO_VERSION | cut -c 3-)
+BUILD_IMAGE := docker.io/library/golang:$(shell echo ${GO_VERSION})
+
+UNAME_S := $(shell uname -s)
+UNAME_M := $(shell uname -m)
+
+GOOS := linux
+HOST_OS := linux
+ifeq ($(UNAME_S),Darwin)
+	HOST_OS := darwin
+endif
+
+ARCH := x86_64
+GOARCH := amd64
+ifeq ($(UNAME_M),arm64)
+	ARCH := aarch64
+	GOARCH := arm64
+endif
+
+BUILD_FLAGS := CGO_ENABLED=0 GOOS=${GOOS} GOARCH=${GOARCH}
+BUILD_CMD := go build -trimpath -ldflags="-X github.com/stackrox/stackrox/scanner/internal/version.Version=$(TAG)" -o image/scanner/bin/scanner ./cmd/scanner
+
+comma:= ,
+empty:=
+space:= $(empty) $(empty)
+
+LOCAL_VOLUME_ARGS := -v $(CURDIR):/src:delegated -v $(GOPATH):/go:delegated
+GOPATH_WD_OVERRIDES := -w /src
+IMAGE_BUILD_FLAGS := -e $(subst $(space),$(comma),$(BUILD_FLAGS)),GOPATH=/go
+
+###############################################################
+###### Binaries we depend on (need to be defined on top) ######
+###############################################################
+
+OSSLS_BIN := $(GOBIN)/ossls
+$(OSSLS_BIN): deps
+	$(SILENT)echo "+ $@"
+	$(SILENT)cd tools/ && go install github.com/stackrox/ossls
+
+#########
+## Tag ##
+#########
+
+.PHONY: tag
+tag:
+	$(SILENT)echo $(TAG)
+
+############
+## Images ##
+############
+
+.PHONY: images
+images: scanner-image db-image
+
+.PHONY: scanner-go-build
+scanner-go-build: deps
+	$(SILENT)echo "+ $@"
+	$(BUILD_FLAGS) $(BUILD_CMD)
```

Comment:
Nitpick: Because you are using recursive make for the local build, we could use the actual go build command here instead of putting it in a variable `BUILD_CMD`. From my perspective, cleaner. But also fine the way it is.